### PR TITLE
BT-8194 some more improvements.

### PR DESCRIPTION
**Pull request by amospaul-bsn**
Reviewer: @3mp4y5 @kenchiang @wilmo119 

Things done:
1) Changing the behaviour to avoid poll 0 in case delta > INT32_MAX 
2) Added more prints and NOTES

### CODE REVIEW

**Comment by 3mp4y5** [comment]
what if we added an API which allowed the caller to set the max timout?

i think that would be more inline with the concerns you've expressed about the implementations...
then the max timeout is explicit .

**Comment by amospaul-bsn** [comment]
There is another way to improve responsiveness. All you need to do is to register a timer with the appropriate period. This will ensure that the poll would choose an appropriate timeout always. This is a hacky way of getting things done.

**Comment by 3mp4y5** [comment]
> This will ensure that the poll would choose an appropriate timeout always

Do you feel comfortable with this approach?

**Comment by amospaul-bsn** [comment]
I would prefer setting the max timeout. That would be very intentional of what we are doing.

**Comment by 3mp4y5** [comment]
> I would prefer setting the max timeout.
Do you mean via a field in ind_soc_init()'s ind_soc_config_t?

Is there pushback from anyone else?

**Comment by kenchiang** [comment]
> Do you mean via a field in ind_soc_init()'s ind_soc_config_t?

I'd suggest a compile time option instead, following the other config in SocketManager.yml.

**Comment by amospaul-bsn** [comment]
> > Do you mean via a field in ind_soc_init()'s ind_soc_config_t?
> 
> I'd suggest a compile time option instead, following the other config in SocketManager.yml.

I think that is a great idea as long as each application is able to choose a value appropriate to them.

File: modules/SocketManager/module/src/socketmanager.c [hunk 1]
```diff
@@ -370,6 +370,11 @@ ind_soc_socket_unregister(int socket_id)
 
 static ind_soc_run_status_t ind_soc_run_status__;
 
+/*
+ * NOTE: using this api to exit ind_soc_select_and_run may not always result in
+ * a timely exit if the registered callbacks are long. Consider using other
```
Comment (kenchiang) [inline modules/SocketManager/module/src/socketmanager.c]: to what does "registered callbacks are long" refer?
do you mean timer events with large values for repeat_time_ms?  or sockets that infrequently are read ready?

Comment (amospaul-bsn) [reply modules/SocketManager/module/src/socketmanager.c]: *registered callbacks have long timer durations.

Comment (kenchiang) [reply modules/SocketManager/module/src/socketmanager.c]: suggest rewording "registered callbacks are long" to "timer events with long times between timer callbacks"

Comment (amospaul-bsn) [reply modules/SocketManager/module/src/socketmanager.c]: sure will do.

File: modules/SocketManager/module/auto/SocketManager.yml [hunk 2]
```diff
@@ -60,6 +60,9 @@ cdefs: &cdefs
 - SOCKETMANAGER_CONFIG_MAX_SOCKETS:
     doc: "Maximum number of sockets supported"
     default: 1024
+- SOCKETMANAGER_CONFIG_MAX_POLL_TIMEOUT:
+    doc: "Restrict the poll time out (in ms) to improve responsiveness. -1 for infinite"
```
Comment (3mp4y5) [inline modules/SocketManager/module/auto/SocketManager.yml]: improve responsiveness is vague.
i think this bounds the max amount of time wait between requesting IND_SOC_RUN_STATUS_EXIT and returning from ind_soc_select_and_run

1024 is sort of odd for a ms value, since its 1.024 seconds... what about 1000?

File: modules/SocketManager/module/inc/SocketManager/socketmanager_config.h [hunk 3]
```diff
@@ -158,6 +158,16 @@
...
+/**
+ * SOCKETMANAGER_CONFIG_MAX_POLL_TIMEOUT
+ *
+ * Restrict the poll time out (in ms) to improve responsiveness. -1 for infinite */
+
+
+#ifndef SOCKETMANAGER_CONFIG_MAX_POLL_TIMEOUT
+#define SOCKETMANAGER_CONFIG_MAX_POLL_TIMEOUT 1024
+#endif
+
```
Comment (3mp4y5) [inline modules/SocketManager/module/inc/SocketManager/socketmanager_config.h]: We ought to get feedback from @kenchiang here ... i though he wanted this hidden behind compile time options, to prevent the new code from getting used if its not specifically requested ...   i would think this defined the value in all cases...

File: modules/SocketManager/module/auto/SocketManager.yml (hunk 2 above)
Comment (kenchiang) [inline modules/SocketManager/module/auto/SocketManager.yml]: How about ""Restrict the poll time out (in ms) to improve responsiveness; set to a negative value for infinite timeout" ?  It seems that poll only cares that the timeout is negative.

File: modules/SocketManager/module/inc/SocketManager/socketmanager_config.h (hunk 3 above)
Comment (kenchiang) [reply modules/SocketManager/module/inc/SocketManager/socketmanager_config.h]: Actually, this code is automatically generated from SocketManager.yml.

File: modules/SocketManager/module/auto/SocketManager.yml (hunk 2 above)
Comment (amospaul-bsn) [reply modules/SocketManager/module/auto/SocketManager.yml]: Sure will update.

Comment (amospaul-bsn) [reply modules/SocketManager/module/auto/SocketManager.yml]: 1000 works too. Yes I also wanted it to be able to perform the infinite poll if explicitly desired.

File: modules/SocketManager/module/inc/SocketManager/socketmanager_config.h (hunk 3 above)
Comment (amospaul-bsn) [reply modules/SocketManager/module/inc/SocketManager/socketmanager_config.h]: This is an auto generated file based on SocketManager.yml

**Comment by kenchiang** [comment]
looks good ; thanks for making the changes.
